src/sentry/event_manager.py

```diff
@@ -1880,6 +1880,101 @@ def _calculate_span_grouping(jobs, projects):
+
+                    try:
+                        short_id = project.next_short_id()
+                    except OperationalError:
+                        metrics.incr(
+                            "next_short_id.timeout",
+                            tags={"platform": "unknown"},
+                        )
+                        sentry_sdk.capture_message("short_id.timeout")
+                        raise HashDiscarded("Timeout when getting next_short_id")
+
+                    # TODO: RATE LIMITER
+                    # from sentry.sentry_metrics.configuration import UseCaseKey
+                    # from sentry.sentry_metrics.indexer.base import KeyCollection
+                    # from sentry.sentry_metrics.indexer.ratelimiters import writes_limiter
+                    # with writes_limiter.check_write_limits(UseCaseKey("performance"), KeyCollection({job.organization.id: {job.organization.name}})) as writes_limiter_state:
+                    #     pass
+
+                    group = Group.objects.create(
+                        project=project,
+                        short_id=short_id,
+                        first_release_id=None,
+                    )
+
+                    is_new = True
+                    is_regression = False
+
+                    span.set_tag("create_group_transaction.outcome", "new_group")
+                    metric_tags["create_group_transaction.outcome"] = "new_group"
+
+                    metrics.incr(
+                        "group.created",
+                        skip_internal=True,
+                        tags={"platform": "unknown"},
+                    )
+
+                    job["group"] = group
+                    job["event"].group = job["group"]
+                    job["is_new"] = is_new
+                    job["is_regression"] = is_regression
+                    continue
+
+            # GROUP EXISTS
+            existing_grouphash = GroupHash.objects.get(project=project, hash=None)
+
+            group = Group.objects.filter(id=existing_grouphash.group_id)
+
+            is_new = False
+            kwargs = {
+                "platform": job["platform"],
+                "message": job["event"].search_message,
+                "culprit": job["culprit"],
+                "logger": job["logger_name"],
+                "level": LOG_LEVELS_MAP.get(job["level"]),
+                "last_seen": job["event"].datetime,
+                "first_seen": job["event"].datetime,
+                "active_at": job["event"].datetime,
+            }
+
+            is_regression = _process_existing_aggregate(
+                group=group, event=job["event"], data=kwargs, release=None
+            )
+
+            # TODO: make groups array
+            # return group IDs we've associated with the transaction
+            job["group"] = group
+            job["event"].group = job["group"]
+            job["is_new"] = is_new
+            job["is_regression"] = is_regression
```

Comment:
We should also default these values if no fingerprint is present in the group at all

Reply:
is this not the same as the code in line `1942`?

Reply:
Your line numbers have moved around a bit so I'm not sure where you were referring to.

The difference is that in the old code, we set the default values for every job. After these changes the default values are only set for jobs that create/update an existing group,  but not jobs that have no fingerprint. It could make sense to default these values at the start so that we know they're always at least set.

Reply:
Got it. uncommented the defaults